cedars/app/ops.py

```diff
@@ -616,8 +616,22 @@ def _adjudicate_annotation(updated_date = False):
             else:
                 session["index"] = session["unreviewed_annotations_index"].index(1)
         elif 1 in session["unreviewed_annotations_index"]:
-            # any unreviewed annotations left?
-            session["index"] = session["unreviewed_annotations_index"].index(1)
+            # Handles the case of entering a date on a note already marked as
+            # adjudicated
+            if updated_date and skip_after_event:
+                session["unreviewed_annotations_index"] = [0] * len(session["unreviewed_annotations_index"])
+                db.set_patient_lock_status(session["patient_id"], False)
+                session.pop("patient_id")
+
+            if db.get_event_date(session["patient_id"]) is not None:
```

Comment:
why is this check required here?

Reply:
This check accounts for the case where an event date was entered on annotation that was already marked as reviewed.